cli/src/commands/start.rs

```diff
@@ -228,11 +228,14 @@ impl Start {
             false => Ok(self
                 .peers
                 .split(',')
-                .flat_map(|ip| match ip.parse::<SocketAddr>() {
-                    Ok(ip) => Some(ip),
-                    Err(e) => {
-                        eprintln!("The IP supplied to --peers ('{ip}') is malformed: {e}");
-                        None
+                .flat_map(|ip_or_hostname| {
+                    let trimmed = ip_or_hostname.trim();
+                    match trimmed.to_socket_addrs() {
+                        Ok(mut addr) => addr.next(),
+                        Err(e) => {
+                            eprintln!("The hostname or IP supplied to --peers ('{trimmed}') is malformed: {e}");
+                            None
+                        }
```

Comment:
this is a little bit convoluted; what we should be doing is first collect the peers (once split by `,`) to a vector, and then call `to_socket_addrs()` on it, and collect it into a `Result<Vec<SocketAddr, _>>`, which would be the final output

Reply:
discussed this online, and it's fine as long as only the 1st resolved IP is of interest and it's documented :+1:

Reply:
Great suggestion (incl offline discussion). I've added a comment to clarify. Moreover, I've change the variable name as you suggested. To make it easier to read the code. :sun_with_face: